Non SSL client is crashing SSL server

**Describe the bug**
If server is running with `$ssl=true`, and a client tries to connect to `ws://...` url, the server stops. (connecting to `wss://...` url works fine)  
Idealy the server should respond with `HTTP 301 Moved Permanently\r\nLocation: wss://...`, and the client follow it. 
Worst case the server should simply reject the client, but definitely not crash.

**How to reproduce**
Start server with `$ssl=true` (and correct ssl context), then connect a client using url starting with: `ws://`

**Logs**
[2025.01.06 10:51:01] [app] [INFO] [server] Starting server on ssl://0.0.0.0:8888. [Context [] Extra {"file":"/test/vendor/phrity/websocket/src/Server.php","line":474,"class":"WebSocket\\Server","callType":"->","function":"createSocketServer"}]
[2025.01.06 10:51:01] [app] [INFO] [server] Server is running [Context [] Extra {"file":"/test/vendor/phrity/websocket/src/Server.php","line":324,"class":"WebSocket\\Server","callType":"->","function":"start"}]
[2025.01.06 10:51:07] [app] [ERROR] [server] Could not accept on socket. [Context [] Extra {"file":"/test/vendor/phrity/websocket/src/Server.php","line":383,"class":"WebSocket\\Server","callType":"->","function":"start"}]
[2025.01.06 10:51:07] [app] [INFO] [server] Server disconnected [Context [] Extra {"file":"/test/vendor/phrity/websocket/src/Server.php","line":459,"class":"WebSocket\\Server","callType":"->","function":"disconnect"}]
[2025.01.06 10:51:07] [app] [ERROR] Could not accept on socket. [Context [] Extra {"file":"/test/src/Command/WebSocketStartCommand.php","line":55,"class":"App\\Command\\WebSocketStartCommand","callType":"->","function":"execute"}]
[2025.01.06 10:51:07] [console] [DEBUG] Command "'websocket:start'" exited with code "1" [Context {"command":"'websocket:start'","code":1} Extra {"file":"/test/vendor/symfony/console/EventListener/ErrorListener.php","line":67,"class":"Symfony\\Component\\Console\\EventListener\\ErrorListener","callType":"->","function":"onConsoleTerminate"}]

**Versions**
* 3.2.1
* PHP 8.2.8

--- a/examples/send.php
+++ b/examples/send.php
@@ -10,7 +10,7 @@
  * Run in console: php examples/send.php <options> <message>
  *
  * Console options:
- *  --uri <uri> : The URI to connect to, default ws://localhost:8000
+ *  --uri <uri> : The URI to connect to, default ws://localhost:80
  *  --opcode <string> : Opcode to send, default 'text'
  *  --debug : Output log data (if logger is available)
  */
@@ -20,7 +20,6 @@ namespace WebSocket;
 require __DIR__ . '/../vendor/autoload.php';
 
 error_reporting(-1);
-
 
 echo "# Send client! [phrity/websocket]\n";
 

--- a/src/Exception/ConnectionFailureException.php
+++ b/src/Exception/ConnectionFailureException.php
@@ -13,8 +13,8 @@ namespace WebSocket\Exception;
  */
 class ConnectionFailureException extends Exception implements ConnectionLevelInterface
 {
-    public function __construct()
+    public function __construct(string|null $message = null)
     {
-        parent::__construct('Connection error');
+        parent::__construct($message ?? 'Connection error');
     }
 }

--- a/src/Http/HttpHandler.php
+++ b/src/Http/HttpHandler.php
@@ -113,7 +113,7 @@ class HttpHandler implements LoggerAwareInterface, Stringable
                 throw new RuntimeException('Could not read Http request.');
             }
             $data .= $buffer;
-        } while (!str_ends_with($data, "\r\n"));
+        } while (!str_ends_with($data, "\n"));
         return trim($data);
     }
 }

--- a/src/Message/Message.php
+++ b/src/Message/Message.php
@@ -75,7 +75,10 @@ abstract class Message implements Stringable
     public function getFrames(int $frameSize = 4096): array
     {
         $frames = [];
-        $split = str_split($this->getPayload(), $frameSize) ?: [''];
+        $split = str_split($this->getPayload(), $frameSize);
+        if (empty($split)) {
+            $split = [''];
+        }
         foreach ($split as $i => $payload) {
             $frames[] = new Frame(
                 $i === 0 ? $this->opcode : 'continuation',

--- a/src/Server.php
+++ b/src/Server.php
@@ -485,6 +485,7 @@ class Server implements LoggerAwareInterface, Stringable
     // Accept connection on socket server
     protected function acceptSocket(SocketServer $socket): void
     {
+        $connection = null;
         try {
             if (!is_null($this->maxConnections) && $this->getConnectionCount() >= $this->maxConnections) {
                 $this->logger->warning("[server] Denied connection, reached max {$this->maxConnections}");
@@ -512,7 +513,8 @@ class Server implements LoggerAwareInterface, Stringable
                 $connection->getHandshakeResponse(),
             ]);
             $this->dispatch('connect', [$this, $connection, $request]);
-        } catch (Exception|StreamException $e) {
+        } catch (Exception | StreamException $e) {
+            /** @var Connection|null $connection */
             if (isset($connection)) {
                 $connection->disconnect();
             }

--- a/src/Server.php
+++ b/src/Server.php
@@ -11,6 +11,7 @@ use InvalidArgumentException;
 use Phrity\Net\{
     SocketServer,
     StreamCollection,
+    StreamException,
     StreamFactory,
     Uri
 };
@@ -23,6 +24,7 @@ use Stringable;
 use Throwable;
 use WebSocket\Exception\{
     CloseException,
+    ConnectionFailureException,
     ConnectionLevelInterface,
     Exception,
     HandshakeException,
@@ -510,12 +512,11 @@ class Server implements LoggerAwareInterface, Stringable
                 $connection->getHandshakeResponse(),
             ]);
             $this->dispatch('connect', [$this, $connection, $request]);
-        } catch (Exception $e) {
+        } catch (Exception|StreamException $e) {
             if (isset($connection)) {
                 $connection->disconnect();
             }
-            $error = "Server failed to accept: {$e->getMessage()}";
-            throw $e;
+            throw new ConnectionFailureException("Server failed to accept: {$e->getMessage()}");
         }
     }
 

